code/modules/research/xenobiology/crossbreeding/warping.dm

```diff
@@ -0,0 +1,1135 @@
+
+
+/* black space rune : will swap out the species of the two next person walking on the rune  */
+
+
+/obj/item/slimecross/warping/black
+	colour = "black"
+	runepath = /obj/effect/warped_rune/blackspace
+	effect_desc = "Will swap the species of the first two humanoids that walk on the rune. Also works on corpses."
+	drawing_time = 100
+
+
+/obj/effect/warped_rune/blackspace
+	icon_state = "cursed_rune"
+	desc = "Your body is the problem, limited, so very very limited."
+	///first person to step on the rune
+	var/mob/living/carbon/human/first_person
+	///second person to step on the rune
+	var/mob/living/carbon/human/second_person
+	///here to check if someone already stepped on the rune
+	var/stepped_on = FALSE
+
+
+/obj/effect/warped_rune/blackspace/Initialize()
+	. = ..()
+	cooldown = 0 //doesn't start on cooldown like most runes
+
+
+///will swap the species of the first two human or human subset that walk on the rune
+/obj/effect/warped_rune/blackspace/Crossed(atom/movable/crossing)
+	. = ..()
+	if(cooldown > world.time) //here to avoid spam/lag
+		to_chat(crossing, "<span class='warning'>The rune needs a little more time before processing your DNA!</span>")
+		return
+
+	if(!istype(crossing,/mob/living/carbon/human))
+		return
+
+	var/mob/living/carbon/human/crosser = crossing
+	if(!stepped_on)
+		first_person = crosser
+		stepped_on = TRUE
+		return
+
+	if(crosser == first_person || crosser.stat == DEAD)
+		return
+
+	second_person = crosser
+	var/first_dna = first_person.dna.species
+	var/second_dna = second_person.dna.species
+	second_person.set_species(first_dna)  //swap the species
+	first_person.set_species(second_dna)
+	stepped_on = FALSE
+	cooldown = max_cooldown + world.time //the default max cooldown is of 10 seconds
+
+/obj/effect/warped_rune/blackspace/Destroy()
+	first_person = null
+	second_person = null
+	return ..()
+
+
+/*Acts as a bunker making anything into the rune immune to outside explosions and gas leaks. doesn't work if the epicenter of the explosion is IN the rune */
+
+
+/obj/item/slimecross/warping/oil
+	colour = "oil"
+	runepath = /obj/effect/warped_rune/oilspace
+	effect_desc = "protects anything on the rune from explosions unless the rune is in the center of the explosion."
+
+
+/obj/effect/warped_rune/oilspace
+	icon_state = "oil_rune"
+	desc = "The world is ending, but we have one last trick up our sleeve, we will survive."
+	///used to remember the oilspace_bunker specific to this rune
+	var/list/bunker_list = list()
+
+
+/obj/effect/oilspace_bunker //we'll surround the rune with these so it "blocks" nearby explosions. Although only the rune itself is 100% protected
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	invisibility = TRUE //no one can see those
+	anchored = TRUE
+	explosion_block = INFINITY
+	CanAtmosPass = ATMOS_PASS_NO //will also try to stop gas from exiting the rune to keep the pressure and air of the tile.Doesn't always work
```

Comment:
might want to ensure that you can't accidentally suffocate departments with these tiles by blocking the atmos vents/entrances with them